File: backend/src/lib/email.js
'use strict';

const nodemailer = require('nodemailer');
const cfg        = require('../config').email;

let _transporter = null;

function getTransporter() {
  if (!_transporter) {
    _transporter = nodemailer.createTransport({
      host:   cfg.host,
      port:   cfg.port,
      secure: cfg.secure,
      auth: {
        user: cfg.username,
        pass: cfg.password,
      },
    });
  }
  return _transporter;
}

/**
 * Gửi email HTML.
 * @param {object} opts
 * @param {string}   opts.to
 * @param {string}   opts.subject
 * @param {string}   opts.html
 * @param {string}  [opts.text]   - plain-text fallback
 */
async function sendMail({ to, subject, html, text }) {
  const info = await getTransporter().sendMail({
    from:    `"${cfg.fromName}" <${cfg.username}>`,
    to,
    subject,
    html,
    text: text || html.replace(/<[^>]*>/g, ''),
  });
  return info;
}

// ── Template helpers ─────────────────────────────────────────────────────────

function tplWelcome({ name }) {
  return {
    subject: 'Chào mừng bạn đến với Tropia!',
    html: `
      <h2>Xin chào ${name}!</h2>
      <p>Cảm ơn bạn đã đăng ký tài khoản Tropia.</p>
      <p>Trải nghiệm mua sắm livestream nông sản tươi ngon nhất Việt Nam ngay hôm nay.</p>
      <br>
      <p>Trân trọng,<br><strong>Đội ngũ Tropia</strong></p>
    `,
  };
}

function tplOrderConfirm({ buyerName, productName, quantity, totalPrice, sessionTitle }) {
  return {
    subject: `Xác nhận đơn hàng – ${productName}`,
    html: `
      <h2>Xác nhận đơn hàng</h2>
      <p>Xin chào <strong>${buyerName}</strong>,</p>
      <p>Đơn hàng của bạn đã được đặt thành công trong buổi live <strong>${sessionTitle}</strong>.</p>
      <table>
        <tr><td><b>Sản phẩm</b></td><td>${productName}</td></tr>
        <tr><td><b>Số lượng</b></td><td>${quantity}</td></tr>
        <tr><td><b>Tổng tiền</b></td><td>${Number(totalPrice).toLocaleString('vi-VN')}₫</td></tr>
      </table>
      <br>
      <p>Trân trọng,<br><strong>Đội ngũ Tropia</strong></p>
    `,
  };
}

function tplPasswordReset({ name, resetUrl }) {
  return {
    subject: 'Đặt lại mật khẩu Tropia',
    html: `
      <h2>Đặt lại mật khẩu</h2>
      <p>Xin chào ${name},</p>
      <p>Nhấn vào liên kết bên dưới để đặt lại mật khẩu của bạn (có hiệu lực trong 15 phút):</p>
      <p><a href="${resetUrl}" style="background:#4CAF50;color:white;padding:10px 20px;text-decoration:none;border-radius:4px;">Đặt lại mật khẩu</a></p>
      <p>Nếu bạn không yêu cầu, hãy bỏ qua email này.</p>
      <br>
      <p>Trân trọng,<br><strong>Đội ngũ Tropia</strong></p>
    `,
  };
}

function tplEmailOtp({ name, otp }) {
  return {
    subject: 'Mã xác thực OTP – Tropia',
    html: `
      <div style="font-family:sans-serif;max-width:480px;margin:0 auto">
        <h2 style="color:#2E7D32">Xác thực tài khoản Tropia</h2>
        <p>Xin chào <strong>${name}</strong>,</p>
        <p>Nhập mã OTP sau vào ứng dụng để kích hoạt tài khoản của bạn:</p>
        <p style="text-align:center;margin:32px 0">
          <span style="display:inline-block;background:#f5f5f5;border:2px dashed #4CAF50;border-radius:12px;padding:16px 40px;font-size:36px;font-weight:900;letter-spacing:10px;color:#2E7D32">${otp}</span>
        </p>
        <p style="color:#888;font-size:13px">Mã có hiệu lực trong <strong>10 phút</strong>. Không chia sẻ mã này với ai.</p>
        <p>Nếu bạn không đăng ký tài khoản Tropia, hãy bỏ qua email này.</p>
        <br>
        <p>Trân trọng,<br><strong>Đội ngũ Tropia</strong></p>
      </div>
    `,
  };
}

/**
 * @param {object} opts
 * @param {string}   opts.buyerName
 * @param {string}   opts.orderId
 * @param {string}   opts.method        - MoMo | ZaloPay | VNPay
 * @param {string}   opts.transId
 * @param {number}   opts.amount        - grand total (đã giảm giá)
 * @param {number}   opts.subtotal      - tổng trước giảm
 * @param {number}   opts.discountAmount
 * @param {Array}    opts.items         - [{ name, quantity, unitPrice }]
 * @param {string}   opts.paidAt        - ISO date string
 */
function tplPaymentSuccess({ buyerName, orderId, method, transId, amount, subtotal, discountAmount, items, paidAt }) {
  const fmt  = n => Number(n || 0).toLocaleString('vi-VN') + ' đ';
  const date = paidAt ? new Date(paidAt).toLocaleString('vi-VN', { timeZone: 'Asia/Ho_Chi_Minh' }) : '';

  const itemRows = (items || []).map(item => `
    <tr>
      <td style="padding:10px 12px;border-bottom:1px solid #f0f0f0;color:#333">${item.name}</td>
      <td style="padding:10px 12px;border-bottom:1px solid #f0f0f0;text-align:center;color:#333">${item.quantity}</td>
      <td style="padding:10px 12px;border-bottom:1px solid #f0f0f0;text-align:right;color:#333">${fmt(item.unitPrice)}</td>
      <td style="padding:10px 12px;border-bottom:1px solid #f0f0f0;text-align:right;font-weight:600;color:#333">${fmt(item.unitPrice * item.quantity)}</td>
    </tr>`).join('');

  const discountRow = discountAmount > 0 ? `
    <tr>
      <td colspan="3" style="padding:8px 12px;text-align:right;color:#555">Giảm giá voucher</td>
      <td style="padding:8px 12px;text-align:right;color:#2E7D32;font-weight:600">-${fmt(discountAmount)}</td>
    </tr>` : '';

  return {
    subject: `${method} – Thanh toán thành công`,
    html: `
<!DOCTYPE html>
<html lang="vi">
<head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1"></head>
<body style="margin:0;padding:0;background:#f5f5f5;font-family:'Segoe UI',Arial,sans-serif">
  <div style="max-width:600px;margin:32px auto;background:#fff;border-radius:12px;overflow:hidden;box-shadow:0 2px 12px rgba(0,0,0,0.08)">

    <!-- Header -->
    <div style="background:linear-gradient(135deg,#2E7D32,#43A047);padding:32px 24px;text-align:center">
      <div style="width:64px;height:64px;background:rgba(255,255,255,0.2);border-radius:50%;margin:0 auto 16px;display:flex;align-items:center;justify-content:center">
        <span style="font-size:32px">✓</span>
      </div>
      <h1 style="color:#fff;margin:0;font-size:24px;font-weight:700">Thanh toán thành công!</h1>
      <p style="color:rgba(255,255,255,0.85);margin:8px 0 0;font-size:14px">${date}</p>
    </div>

    <!-- Body -->
    <div style="padding:28px 24px">
      <p style="margin:0 0 20px;color:#444;font-size:15px">Xin chào <strong>${buyerName}</strong>,</p>
      <p style="margin:0 0 24px;color:#444;font-size:14px">
        Đơn hàng <strong style="color:#2E7D32">${orderId}</strong> đã được thanh toán thành công qua <strong>${method}</strong>.
      </p>

      <!-- Transaction info -->
      <table style="width:100%;border-collapse:collapse;margin-bottom:24px;background:#f9f9f9;border-radius:8px;overflow:hidden">
        <tr>
          <td style="padding:10px 16px;color:#666;font-size:13px;width:40%">Mã giao dịch</td>
          <td style="padding:10px 16px;color:#333;font-weight:600;font-size:13px">${transId}</td>
        </tr>
        <tr style="background:#f1f1f1">
          <td style="padding:10px 16px;color:#666;font-size:13px">Phương thức</td>
          <td style="padding:10px 16px;color:#333;font-weight:600;font-size:13px">${method}</td>
        </tr>
        ${date ? `<tr>
          <td style="padding:10px 16px;color:#666;font-size:13px">Thời gian</td>
          <td style="padding:10px 16px;color:#333;font-size:13px">${date}</td>
        </tr>` : ''}
      </table>

      <!-- Items table -->
      <h3 style="margin:0 0 12px;color:#333;font-size:15px;font-weight:700">Chi tiết sản phẩm</h3>
      <table style="width:100%;border-collapse:collapse;border:1px solid #eee;border-radius:8px;overflow:hidden;margin-bottom:16px">
        <thead>
          <tr style="background:#f5f5f5">
            <th style="padding:10px 12px;text-align:left;font-size:13px;color:#555;font-weight:600">Sản phẩm</th>
            <th style="padding:10px 12px;text-align:center;font-size:13px;color:#555;font-weight:600">SL</th>
            <th style="padding:10px 12px;text-align:right;font-size:13px;color:#555;font-weight:600">Đơn giá</th>
            <th style="padding:10px 12px;text-align:right;font-size:13px;color:#555;font-weight:600">Thành tiền</th>
          </tr>
        </thead>
        <tbody>
          ${itemRows}
          ${discountRow}
          <tr style="background:#fff8e1">
            <td colspan="3" style="padding:12px;text-align:right;font-weight:700;color:#333;font-size:15px">Tổng thanh toán</td>
            <td style="padding:12px;text-align:right;font-weight:800;color:#E65100;font-size:18px">${fmt(amount)}</td>
          </tr>
        </tbody>
      </table>

      <p style="margin:24px 0 0;color:#888;font-size:13px;text-align:center">
        Cảm ơn bạn đã mua sắm tại <strong style="color:#2E7D32">Tropia</strong>!<br>
        Đơn hàng sẽ được giao trong 1–3 ngày làm việc.
      </p>
    </div>

    <!-- Footer -->
    <div style="background:#f9f9f9;padding:16px 24px;text-align:center;border-top:1px solid #eee">
      <p style="margin:0;font-size:12px;color:#aaa">© 2026 Tropia – Nông sản tươi ngon mỗi ngày</p>
    </div>
  </div>
</body>
</html>`,
  };
}

module.exports = { sendMail, tplWelcome, tplOrderConfirm, tplPasswordReset, tplEmailOtp, tplPaymentSuccess };

File: backend/internal/notify/email.go
package notify

import (
	"crypto/tls"
	"fmt"
	"net/smtp"
	"strings"
	"time"
)

type EmailConfig struct {
	Host     string
	Port     int
	Username string
	Password string
	FromName string
	Secure   bool
}

type Email struct{ cfg EmailConfig }

func NewEmail(cfg EmailConfig) *Email { return &Email{cfg: cfg} }

func (e *Email) Send(to, subject, htmlBody string) error {
	from := fmt.Sprintf("%s <%s>", e.cfg.FromName, e.cfg.Username)
	msg := []byte(
		"From: " + from + "\r\n" +
			"To: " + to + "\r\n" +
			"Subject: " + subject + "\r\n" +
			"MIME-Version: 1.0\r\n" +
			"Content-Type: text/html; charset=UTF-8\r\n\r\n" +
			htmlBody,
	)
	addr := fmt.Sprintf("%s:%d", e.cfg.Host, e.cfg.Port)
	auth := smtp.PlainAuth("", e.cfg.Username, e.cfg.Password, e.cfg.Host)

	if e.cfg.Secure {
		// SMTPS (port 465)
		conn, err := tls.Dial("tcp", addr, &tls.Config{ServerName: e.cfg.Host})
		if err != nil {
			return err
		}
		client, err := smtp.NewClient(conn, e.cfg.Host)
		if err != nil {
			return err
		}
		defer client.Quit()
		if err := client.Auth(auth); err != nil {
			return err
		}
		if err := client.Mail(e.cfg.Username); err != nil {
			return err
		}
		if err := client.Rcpt(to); err != nil {
			return err
		}
		w, err := client.Data()
		if err != nil {
			return err
		}
		if _, err := w.Write(msg); err != nil {
			return err
		}
		return w.Close()
	}
	// STARTTLS (587) - smtp.SendMail handles upgrade
	return smtp.SendMail(addr, auth, e.cfg.Username, []string{to}, msg)
}

// ============================================================================
// Email templates — Tropia brand
// ============================================================================
//
// Brand colors:
//   primary green = #16A34A   (CTA buttons, success states)
//   dark green    = #0D6B33   (headers, footers)
//   accent orange = #EA580C   (totals, sale prices)
//   light bg      = #F6FBF6   (card backgrounds)
//
// Tagline: "Tropia – Nông sản tươi ngon mỗi ngày"
//
// All templates render inline-styled HTML (max email-client compat) inside
// a 600px centered card with rounded corners.

const (
	brandPrimary = "#16A34A"
	brandDark    = "#0D6B33"
	brandAccent  = "#EA580C"
	brandLight   = "#F6FBF6"
	brandBorder  = "#E5E7EB"
	brandMuted   = "#6B7280"
)

// envelope wraps an arbitrary inner HTML fragment in the standard Tropia card.
func envelope(inner string) string {
	year := time.Now().Year()
	return fmt.Sprintf(`<!doctype html>
<html lang="vi">
  <body style="margin:0;padding:24px 12px;background:#F3F4F6;font-family:'Segoe UI',Arial,sans-serif;color:#111;">
    <table role="presentation" cellpadding="0" cellspacing="0" border="0"
           style="max-width:600px;margin:0 auto;background:#FFFFFF;border-radius:12px;overflow:hidden;box-shadow:0 1px 3px rgba(0,0,0,0.08)">
      <tr><td>%s</td></tr>
      <tr><td style="padding:20px 24px;border-top:1px solid %s;text-align:center;color:%s;font-size:12px;">
        © %d Tropia – Nông sản tươi ngon mỗi ngày
      </td></tr>
    </table>
  </body>
</html>`, inner, brandBorder, brandMuted, year)
}

// ── OTP ─────────────────────────────────────────────────────────────────────

func TmplOTP(name, otp string) (string, string) {
	subject := "Mã xác thực OTP – Tropia"
	body := envelope(fmt.Sprintf(`
		<div style="padding:32px 24px;text-align:center;">
		  <h1 style="margin:0 0 24px 0;color:%s;font-size:22px;font-weight:700;">Xác thực tài khoản Tropia</h1>
		  <p style="margin:0 0 16px 0;font-size:15px;text-align:left;">
		    Xin chào <strong>%s</strong>,
		  </p>
		  <p style="margin:0 0 24px 0;font-size:15px;text-align:left;">
		    Nhập mã OTP sau vào ứng dụng để kích hoạt tài khoản của bạn:
		  </p>
		  <div style="border:2px dashed %s;border-radius:12px;padding:24px;margin:0 0 24px 0;">
		    <div style="font-size:36px;letter-spacing:12px;font-weight:700;color:%s;">%s</div>
		  </div>
		  <p style="margin:0 0 12px 0;color:%s;font-size:13px;">
		    Mã có hiệu lực trong <strong>10 phút</strong>. Không chia sẻ mã này với ai.
		  </p>
		  <p style="margin:24px 0 0 0;color:%s;font-size:13px;text-align:left;">
		    Nếu bạn không đăng ký tài khoản Tropia, hãy bỏ qua email này.
		  </p>
		  <p style="margin:16px 0 0 0;color:%s;font-size:13px;text-align:left;">
		    Trân trọng,<br/><strong>Đội ngũ Tropia</strong>
		  </p>
		</div>`,
		brandDark, name, brandPrimary, brandPrimary, otp, brandMuted, brandMuted, brandMuted))
	return subject, body
}

// ── Welcome ─────────────────────────────────────────────────────────────────

func TmplWelcome(name string) (string, string) {
	subject := "Chào mừng đến với Tropia 🍎"
	body := envelope(fmt.Sprintf(`
		<div style="background:linear-gradient(135deg,%s 0%%,%s 100%%);padding:40px 24px;text-align:center;color:#fff;">
		  <div style="font-size:48px;line-height:1;margin-bottom:8px;">🌿</div>
		  <h1 style="margin:0;font-size:24px;font-weight:700;">Chào mừng đến với Tropia!</h1>
		  <p style="margin:8px 0 0 0;opacity:0.9;font-size:14px;">Nông sản tươi ngon, giao tận nhà</p>
		</div>
		<div style="padding:32px 24px;">
		  <p style="margin:0 0 16px 0;font-size:15px;">Xin chào <strong>%s</strong>,</p>
		  <p style="margin:0 0 16px 0;font-size:15px;line-height:1.6;">
		    Cảm ơn bạn đã tham gia Tropia — nền tảng mua sắm thực phẩm tươi sống kết hợp livestream
		    bán hàng. Khám phá ngay các tính năng:
		  </p>
		  <ul style="margin:0 0 24px 0;padding-left:20px;font-size:15px;line-height:1.8;">
		    <li>🛒 Mua trái cây, rau củ, hải sản tươi mỗi ngày</li>
		    <li>📺 Xem livestream từ các nhà cung cấp uy tín</li>
		    <li>🎁 Voucher độc quyền chỉ có trong buổi live</li>
		    <li>🚚 Giao hàng nhanh trong 1–3 ngày</li>
		  </ul>
		  <p style="margin:16px 0 0 0;color:%s;font-size:13px;">
		    Trân trọng,<br/><strong>Đội ngũ Tropia</strong>
		  </p>
		</div>`, brandPrimary, brandDark, name, brandMuted))
	return subject, body
}

// ── Password reset ──────────────────────────────────────────────────────────

func TmplPasswordReset(name, link string) (string, string) {
	subject := "Đặt lại mật khẩu Tropia"
	body := envelope(fmt.Sprintf(`
		<div style="padding:32px 24px;">
		  <h1 style="margin:0 0 16px 0;color:%s;font-size:22px;font-weight:700;">Đặt lại mật khẩu</h1>
		  <p style="margin:0 0 16px 0;font-size:15px;">Xin chào <strong>%s</strong>,</p>
		  <p style="margin:0 0 24px 0;font-size:15px;line-height:1.6;">
		    Chúng tôi nhận được yêu cầu đặt lại mật khẩu cho tài khoản của bạn.
		    Click nút bên dưới để đặt lại (liên kết có hiệu lực trong <strong>1 giờ</strong>):
		  </p>
		  <div style="text-align:center;margin:0 0 24px 0;">
		    <a href="%s"
		       style="display:inline-block;padding:14px 32px;background:%s;color:#fff;text-decoration:none;border-radius:8px;font-weight:600;font-size:15px;">
		      Đặt lại mật khẩu
		    </a>
		  </div>
		  <p style="margin:0 0 12px 0;color:%s;font-size:13px;">
		    Hoặc copy đường dẫn này vào trình duyệt:
		  </p>
		  <p style="margin:0 0 24px 0;word-break:break-all;color:%s;font-size:12px;background:%s;padding:12px;border-radius:6px;">
		    %s
		  </p>
		  <p style="margin:16px 0 0 0;color:%s;font-size:13px;">
		    Nếu bạn không yêu cầu đặt lại mật khẩu, hãy bỏ qua email này.
		    Mật khẩu hiện tại của bạn vẫn an toàn.
		  </p>
		  <p style="margin:16px 0 0 0;color:%s;font-size:13px;">
		    Trân trọng,<br/><strong>Đội ngũ Tropia</strong>
		  </p>
		</div>`, brandDark, name, link, brandPrimary, brandMuted, brandDark, brandLight, link, brandMuted, brandMuted))
	return subject, body
}

// ── Order confirm ───────────────────────────────────────────────────────────

func TmplOrderConfirm(name, orderID string, total int) (string, string) {
	subject := "Xác nhận đơn hàng – Tropia"
	body := envelope(fmt.Sprintf(`
		<div style="background:%s;padding:32px 24px;text-align:center;color:#fff;">
		  <div style="font-size:48px;line-height:1;margin-bottom:8px;">📦</div>
		  <h1 style="margin:0;font-size:22px;font-weight:700;">Đơn hàng đã được tiếp nhận</h1>
		  <p style="margin:8px 0 0 0;opacity:0.9;font-size:13px;">%s</p>
		</div>
		<div style="padding:32px 24px;">
		  <p style="margin:0 0 16px 0;font-size:15px;">Xin chào <strong>%s</strong>,</p>
		  <p style="margin:0 0 24px 0;font-size:15px;line-height:1.6;">
		    Đơn hàng <span style="color:%s;font-weight:600;">%s</span> đã được Tropia tiếp nhận.
		    Tổng giá trị: <strong style="color:%s;">%s ₫</strong>
		  </p>
		  <p style="margin:0 0 8px 0;font-size:14px;color:%s;">
		    Chúng tôi sẽ xử lý và giao hàng đến bạn trong <strong>1–3 ngày làm việc</strong>.
		  </p>
		  <p style="margin:16px 0 0 0;color:%s;font-size:13px;">
		    Trân trọng,<br/><strong>Đội ngũ Tropia</strong>
		  </p>
		</div>`,
		brandPrimary, time.Now().Format("15:04:05 02/01/2006"),
		name, brandPrimary, orderID, brandAccent, formatVND(total),
		brandMuted, brandMuted))
	return subject, body
}

// ── Payment success ─────────────────────────────────────────────────────────

// PaymentItem is one line on the payment receipt.
type PaymentItem struct {
	Name     string
	Quantity int
	Price    int // VND per item
}

// TmplPaymentSuccess matches the screenshot layout: hero check + transaction
// table + itemized table + (voucher row) + grand total. Pass an empty
// `items` slice if you only want the summary (the table is then omitted).
func TmplPaymentSuccess(in PaymentSuccessInput) (string, string) {
	subject := "Thanh toán thành công – Tropia"

	// Build product rows
	var rowsHTML strings.Builder
	for _, it := range in.Items {
		fmt.Fprintf(&rowsHTML, `
		    <tr>
		      <td style="padding:12px;border-bottom:1px solid %s;">%s</td>
		      <td style="padding:12px;border-bottom:1px solid %s;text-align:center;">%d</td>
		      <td style="padding:12px;border-bottom:1px solid %s;text-align:right;">%s ₫</td>
		      <td style="padding:12px;border-bottom:1px solid %s;text-align:right;font-weight:600;">%s ₫</td>
		    </tr>`,
			brandBorder, escapeHTML(it.Name),
			brandBorder, it.Quantity,
			brandBorder, formatVND(it.Price),
			brandBorder, formatVND(it.Price*it.Quantity))
	}

	// Voucher row (only if discount > 0)
	voucherRow := ""
	if in.DiscountAmount > 0 {
		voucherRow = fmt.Sprintf(`
		    <tr>
		      <td colspan="3" style="padding:12px;text-align:right;color:%s;">Giảm giá voucher</td>
		      <td style="padding:12px;text-align:right;color:%s;font-weight:600;">-%s ₫</td>
		    </tr>`, brandMuted, brandPrimary, formatVND(in.DiscountAmount))
	}

	// Items section
	itemsSection := ""
	if len(in.Items) > 0 {
		itemsSection = fmt.Sprintf(`
		  <h3 style="margin:24px 0 12px 0;font-size:16px;color:%s;">Chi tiết sản phẩm</h3>
		  <table role="presentation" cellpadding="0" cellspacing="0" border="0"
		         style="width:100%%;border-collapse:collapse;font-size:14px;">
		    <tr style="background:%s;color:%s;font-weight:600;">
		      <th align="left"   style="padding:10px 12px;">Sản phẩm</th>
		      <th align="center" style="padding:10px 12px;">SL</th>
		      <th align="right"  style="padding:10px 12px;">Đơn giá</th>
		      <th align="right"  style="padding:10px 12px;">Thành tiền</th>
		    </tr>
		    %s
		    %s
		    <tr style="background:#FFF7ED;">
		      <td colspan="3" style="padding:14px 12px;text-align:right;font-weight:700;">Tổng thanh toán</td>
		      <td style="padding:14px 12px;text-align:right;color:%s;font-size:18px;font-weight:700;">%s ₫</td>
		    </tr>
		  </table>`,
			brandDark, brandLight, brandDark, rowsHTML.String(), voucherRow,
			brandAccent, formatVND(in.TotalPrice))
	}

	body := envelope(fmt.Sprintf(`
		<div style="background:linear-gradient(135deg,%s 0%%,%s 100%%);padding:32px 24px;text-align:center;color:#fff;">
		  <div style="width:56px;height:56px;margin:0 auto 12px auto;border-radius:50%%;background:#FFFFFF;display:flex;align-items:center;justify-content:center;">
		    <span style="color:%s;font-size:32px;font-weight:700;line-height:56px;">✓</span>
		  </div>
		  <h1 style="margin:0;font-size:24px;font-weight:700;">Thanh toán thành công!</h1>
		  <p style="margin:8px 0 0 0;opacity:0.95;font-size:13px;">%s</p>
		</div>
		<div style="padding:32px 24px;">
		  <p style="margin:0 0 16px 0;font-size:15px;">Xin chào <strong>%s</strong>,</p>
		  <p style="margin:0 0 16px 0;font-size:15px;line-height:1.6;">
		    Đơn hàng <span style="color:%s;font-weight:600;">%s</span> đã được thanh toán thành công
		    qua <strong>%s</strong>.
		  </p>
		  <table role="presentation" cellpadding="0" cellspacing="0" border="0"
		         style="width:100%%;border-collapse:collapse;font-size:14px;margin:16px 0;">
		    <tr><td style="padding:10px 12px;background:%s;color:%s;width:40%%;">Mã giao dịch</td>
		        <td style="padding:10px 12px;background:%s;font-weight:600;">%s</td></tr>
		    <tr><td style="padding:10px 12px;color:%s;">Phương thức</td>
		        <td style="padding:10px 12px;font-weight:600;">%s</td></tr>
		    <tr><td style="padding:10px 12px;background:%s;color:%s;">Thời gian</td>
		        <td style="padding:10px 12px;background:%s;font-weight:600;">%s</td></tr>
		  </table>
		  %s
		  <p style="margin:24px 0 0 0;text-align:center;font-size:14px;">
		    Cảm ơn bạn đã mua sắm tại <strong style="color:%s;">Tropia!</strong>
		  </p>
		  <p style="margin:8px 0 0 0;text-align:center;color:%s;font-size:13px;">
		    Đơn hàng sẽ được giao trong 1–3 ngày làm việc.
		  </p>
		</div>`,
		brandPrimary, brandDark, brandPrimary,
		time.Now().Format("15:04:05 02/01/2006"),
		in.Name, brandPrimary, in.OrderID, in.Method,
		brandLight, brandMuted, brandLight, in.TransactionID,
		brandMuted, in.Method,
		brandLight, brandMuted, brandLight, time.Now().Format("15:04:05 02/01/2006"),
		itemsSection,
		brandPrimary, brandMuted))
	return subject, body
}

// PaymentSuccessInput bundles everything the receipt email needs.
type PaymentSuccessInput struct {
	Name           string        // recipient display name
	OrderID        string        // UUID or human-friendly id
	Method         string        // MoMo / VNPay / ZaloPay
	TransactionID  string        // gateway-specific txn id
	TotalPrice     int           // final amount after discount (VND)
	DiscountAmount int           // 0 = no voucher line
	Items          []PaymentItem // empty → omit product table
}

// ── helpers ─────────────────────────────────────────────────────────────────

// formatVND turns 21250 → "21.250".
func formatVND(n int) string {
	neg := n < 0
	if neg {
		n = -n
	}
	s := fmt.Sprintf("%d", n)
	var out strings.Builder
	if neg {
		out.WriteByte('-')
	}
	// Insert dots every 3 digits from the right.
	rem := len(s) % 3
	if rem == 0 {
		rem = 3
	}
	out.WriteString(s[:rem])
	for i := rem; i < len(s); i += 3 {
		out.WriteByte('.')
		out.WriteString(s[i : i+3])
	}
	return out.String()
}

// escapeHTML — minimal, enough for product names in receipts.
func escapeHTML(s string) string {
	r := strings.NewReplacer(
		"&", "&amp;",
		"<", "&lt;",
		">", "&gt;",
		`"`, "&quot;",
	)
	return r.Replace(s)
}
